server/middleware/courses.coffee

```diff
@@ -82,3 +86,60 @@ module.exports =
     results = yield database.viewSearch(dbq, req)
     results = Course.sortCourses results
     res.send(results)
+
+  postPatch: wrap (req, res) ->
+    # TODO: Generalize this and use for other models, once this has been put through its paces
+    course = yield database.getDocFromHandle(req, Course)
+    if not course
+      throw new errors.NotFound('Course not found.')
+      
+    originalDelta = req.body.delta
+    originalCourse = course.toObject()
+    changedCourse = _.cloneDeep(course.toObject(), (value) -> 
+      return value if value instanceof mongoose.Types.ObjectId
+      return value if value instanceof Date
+      return undefined
+    )
+    jsondiffpatch.patch(changedCourse, originalDelta)
+    
+    # normalize the delta because in tests, changes to patches would sneak in and cause false positives
+    # TODO: Figure out a better system. Perhaps submit a series of paths? I18N Edit Views already use them for their rows.
+    normalizedDelta = jsondiffpatch.diff(originalCourse, changedCourse)
+    normalizedDelta = _.pick(normalizedDelta, _.keys(originalDelta))
+    reasonDidNotPatch = undefined
+
+    validation = tv4.validateMultiple(changedCourse, Course.jsonSchema)
+    if not validation.valid
+      reasonDidNotPatch = 'Did not pass json schema.'
+    else if not isJustFillingTranslations(normalizedDelta)
+      reasonDidNotPatch = 'Adding to existing translations.'
+    else
+      course.set(changedCourse)
+      yield course.save()
+      
+    patch = new Patch(req.body)
+    patch.set({
+      target: {
+        collection: 'course'
+        id: course._id
+        original: course._id
+      }
+      creator: req.user._id
+      status: if reasonDidNotPatch then 'pending' else 'accepted'
+      created: new Date().toISOString()
+      reasonDidNotPatch
+    })
+    database.validateDoc(patch)
+
+    if reasonDidNotPatch
+      yield course.update({ $addToSet: { patches: patch._id }})
+      patches = course.get('patches') or []
+      patches.push patch._id
+      course.set({patches})
+    yield patch.save()
```

Comment:
`reasonDidNotPatch` is a bit confusingly named here — it looks like whenever `reasonDidNotPatch` is set, we save a patch to the list of patches, but whenever it's not set, we save the course directly. I suspect "patch" is used as a verb "apply patch directly instead of saving it as a pending patch"?